bool flag
---
bool is_inited